pom.xml

```diff
@@ -83,7 +83,8 @@
                             !org.apache.tools.ant,!org.joda.convert,!org.objectweb.asm,!sun.misc,*
                         </Import-Package>
                         <!-- Embed the necessary dependencies -->
-                        <Embed-Dependency>*;scope=compile</Embed-Dependency>
+                        <!-- The `inline=true` is important, don't remove it unless you want cryptic problems such as servlets that suddenly stop resolving without explanation -->
+                        <Embed-Dependency>*;scope=compile;inline=true</Embed-Dependency>
```

Comment:
I'm curious about this. The only noticeable difference for me is the way the dependencies are packaged. With `inline=true` everything is just expanded in the jar file, while the opposite makes it neatly arranged in jar files (except for our own code) with the appropriate entries in the manifest files. 

Are we sure this was causing servlets to stop resolving? If so, I don't mind keeping `inline=true` as long as we assume it as tech debt to fix down the line.

Reply:
I am 100% sure this is the cause.

I am 30% sure that I know the reason why, with a lot of speculation and hand-waving involved.

Remember that we needed to do a lot of work to manually locate the ruby gems necessary in order to instantiate an Asciidoctor instance. Because of the way Asciidoctorj is written, we can only do that successfully with `inline=true`.

My guess is that when Sling is trying to figure out what servlets exist, it has to instantiate a copy of the servlet in order to inspect it. (This is where things get reeeally hand-wavy). When it tries to do that without `inline=true`, the servlet fails to initialize due to Asciidoctor throwing some exception and as a result Sling pretends it doesn't even exist.

Thus when the user requests anything.preview, Sling sees it has 0 servlets registered to handle .preview requests and throws a 404.

Giant hole in this theory: none of our rendering servlet loggers were printing anything, not even the ones before we try and get an Asciidoctor instance.

I'd love to know what's really going on here, but I can only guess for now. "Too much magic" and all that.

Reply:
I think you are right... it's exactly that. I wonder if we can either a. extract the files from a jar file, or 2. just inline that specific dependency. As I said, for the time being we should take this one as tech debt, and try to figure it out later.